src/sqlfluff/utils/reflow/helpers.py

```diff
@@ -25,3 +31,21 @@ def pretty_segment_name(segment: BaseSegment) -> str:
         # Reference other segments just by their type.
         # (With underscores as spaces)
         return segment.get_type().replace("_", " ")
+
+
+def deduce_line_indent(raw_segment: RawSegment, root_segment: BaseSegment) -> str:
+    """Given a raw segment, deduce the indent of it's line."""
+    seg_idx = root_segment.raw_segments.index(raw_segment)
+    indent_seg = None
+    for seg in root_segment.raw_segments[seg_idx::-1]:
```

Comment:
Oh interesting! Great idea.

Reply:
Ok - so because we're not starting at the end of the whole series, `reversed()` isn't a great solution - but we can do something with `range()` and a lookup to approximate the same effect.